tests/components/homekit_controller/common.py

```diff
@@ -88,16 +90,19 @@ def __init__(self, hass, entity_id, pairing, accessory):
                 char_name = CharacteristicsTypes.get_short(char.type)
                 self.characteristics[(service_name, char_name)] = char
 
-    async def get_state(self):
-        """Trigger an entity update() and return the current entity state."""
-        await self.hass.helpers.entity_component.async_update_entity(
-            self.entity_id)
+    async def poll_and_get_state(self):
+        """Trigger a time based poll and return the current entity state."""
+        next_update = dt_util.utcnow() + timedelta(seconds=60)
```

Comment:
Make sure to patch `dt_util.utcnow` in the test too to take complete control of time.

Reply:
When I do that it doesn’t trigger a poll for some reason. It only works when I don’t patch it.

Reply:
Make sure to patch in the correct place, and also watch for any throttle. Don't remember if that is in play for this component.

Reply:
Make sure to patch time already before setting up the component. It's when the entity is added that the clock starts.